synapse/storage/state.py

```diff
@@ -1082,6 +1133,12 @@ def __init__(self, db_conn, hs):
             columns=["state_key"],
             where_clause="type='m.room.member'",
         )
+        self.register_background_index_update(
+            self.EVENT_STATE_GROUP_INDEX_UPDATE_NAME,
+            index_name="event_to_state_groups_sg_index",
```

Comment:
worth remembering that we already have this on matrix.org, so would ideally not spend ages building another copy of the index.